cmd/key-helper/importKey.go

```diff
@@ -55,7 +56,11 @@ func importKeyCommand() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			if err := importKeyToTransit(ctx, vaultClient, ciphertext, keyName); err != nil {
+			keyType, vaultRSAOAEPHashFunction, err := checkKeyType(keyPath)
```

Comment:
IIUC this variable naming (`vaultRSAOAEPHashFunction`) is incorrect. While you _do_ pass it to `import`, for the BYOK stage, it is determined by a mapping of `key algorithm->hash` which aligns with [notary's spec](https://github.com/notaryproject/specifications/blob/main/specs/signature-specification.md#signature-algorithm-requirements) and isn't set during the above `wrapPrivateKey` (and thus, isn't used for the BYOK's RSA-OAEP encryption stage).

I don't think this flag is necessary either (due to it not being used in the earlier call), and further, I don't think it'll work (it'd break the parsing on the server, preventing the import, IMO, _because_ it wasn't used and instead artificially determined). I'd remove it and the associated code.

Reply:
See old line 129, new line 134 for where the RSA-OAEP stage is hard-coded to use SHA-256. There is no benefit in making this configurable and I'd avoid doing so. :-)

Reply:
ohh that was a leftover :D I wasn't sure why transit was protesting against sent signature, and later I have discovered that was a `signer.go` issue :)